Unable to compile against target i686-unknown-linux-musl

I'm using rustc 1.61.0, and RustPython on git devel (currently f84b2896f1c9). I'm running on Debian Unstable, on an amd64.

Was prompted by the compile to install musl-related compile utilities, so I have these versions installed from the Debian Unstable repo:

1. musl-dev 1.2.3-1
2. musl-tools 1.2.3-1

This command works as demonstrated:

`cargo run --release demo.py`

This command fails during compilation:

`cargo run --target=i686-unknown-linux-musl --release demo.py`

The error is during this compile step:

`Compiling rustpython-compiler v0.1.2 (/home/david/dev/misc/github/RustPython/compiler/porcelain)`

And there are a bunch of warnings like this:


```
warning: use of deprecated type alias `libc::time_t`: This type is changed to 64-bit in musl 1.2.0, we'll follow that change in the future release. See #1848 for more info.
   --> vm/src/stdlib/time.rs:499:48
    |
499 |             tv_sec: time / (SEC_TO_NS as libc::time_t),
```

But the actual error which terminates the compilation is this:

```
error[E0063]: missing fields `sched_ss_init_budget`, `sched_ss_low_priority`, `sched_ss_max_repl` and 1 other field in initializer of `sched_param`
   --> vm/src/stdlib/posix.rs:571:16
    |
571 |             Ok(libc::sched_param { sched_priority })
    |                ^^^^^^^^^^^^^^^^^ missing `sched_ss_init_budget`, `sched_ss_low_priority`, `sched_ss_max_repl` and 1 other field

For more information about this error, try `rustc --explain E0063`.
warning: `rustpython-vm` (lib) generated 9 warnings
```
...

For further context / background:

What I'm attempting to do is to get a recent python version over on the really ancient debian4 environment, which is also running in a 32-bit environment. The python version shipping there is 2.4.

At the moment I'm attempting to find a more recent alternative. I've seen that basic rust "hello world" cross-compiles work. But most other languages and ecosystems I've tested, no longer target or compile that really ancient environment. Even go-lang binaries targeting linux 32-bit fail with runtime panics on the environment, probably due to needing underlying linux system calls to be present, even though there is no actual libc dependency and the target binary is static. Target platform linux version is 2.6.30-1-686, and gcc version is 4.1.1

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -89,8 +89,8 @@ jobs:
         with:
           target: i686-unknown-linux-gnu
 
-      - name: Install gcc-multilib
-        run: sudo apt-get update && sudo apt-get install gcc-multilib
+      - name: Install gcc-multilib and musl-tools
+        run: sudo apt-get update && sudo apt-get install gcc-multilib musl-tools
       - name: Check compilation for x86 32bit
         uses: actions-rs/cargo@v1
         with:
@@ -106,6 +106,16 @@ jobs:
         with:
           command: check
           args: --target aarch64-linux-android
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          target: i686-unknown-linux-musl
+
+      - name: Check compilation for musl
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --target i686-unknown-linux-musl
 
       - uses: actions-rs/toolchain@v1
         with:

--- a/vm/src/stdlib/posix.rs
+++ b/vm/src/stdlib/posix.rs
@@ -556,6 +556,7 @@ pub mod module {
             target_os = "freebsd",
             target_os = "android"
         ))]
+        #[cfg(not(target_env = "musl"))]
         fn try_to_libc(&self, vm: &VirtualMachine) -> PyResult<libc::sched_param> {
             use crate::AsObject;
             let priority_class = self.sched_priority.class();
@@ -625,6 +626,7 @@ pub mod module {
         target_os = "freebsd",
         target_os = "android"
     ))]
+    #[cfg(not(target_env = "musl"))]
     #[pyfunction]
     fn sched_setscheduler(args: SchedSetschedulerArgs, vm: &VirtualMachine) -> PyResult<i32> {
         let libc_sched_param = args.sched_param_obj.try_to_libc(vm)?;
@@ -675,6 +677,7 @@ pub mod module {
         target_os = "freebsd",
         target_os = "android"
     ))]
+    #[cfg(not(target_env = "musl"))]
     #[pyfunction]
     fn sched_setparam(args: SchedSetParamArgs, vm: &VirtualMachine) -> PyResult<i32> {
         let libc_sched_param = args.sched_param_obj.try_to_libc(vm)?;

